### PR TITLE
fix dot-prop and serialize-javascript high severity issues

### PR DESCRIPTION
**Pull request by TimoPeraza**
### Description:
* Fix dot-prop and serialize-javascript security problem

#### serialize-javascript
[CVE-2020-7660](https://github.com/advisories/GHSA-hxcc-f52p-wc94) **high severity**

**Vulnerable versions**: < 3.1.0
**Patched version**: 3.1.0

serialize-javascript prior to 3.1.0 allows remote attackers to inject arbitrary code via the function "deleteFunctions" within "index.js".

An object such as {"foo": /1"/, "bar": "a\"@__R-<UID>-0__@"} was serialized as {"foo": /1"/, "bar": "a\/1"/}, which allows an attacker to escape the bar key. This requires the attacker to control the values of both foo and bar and guess the value of <UID>. The UID has a keyspace of approximately 4 billion making it a realistic network attack.

The following proof-of-concept calls console.log() when the running eval():
eval('('+ serialize({"foo": /1" + console.log(1)/i, "bar": '"@__R-<UID>-0__@'}) + ')');

#### dot-prop

[CVE-2020-8116](https://github.com/advisories/GHSA-ff7x-qrg7-qggm) **high severity**

**Vulnerable versions**: < 5.1.1
**Patched version**: 5.1.1

Prototype pollution vulnerability in dot-prop npm package version 5.1.0 and earlier allows an attacker to add arbitrary properties to JavaScript language constructs such as objects.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,10 +8,14 @@
     "@rails/ujs": "^6.0.0-alpha",
     "@rails/webpacker": "^4.2.2",
     "arctic_admin": "3.2.0",
+    "serialize-javascript": "^4.0.0",
```
Comment (santib) [inline package.json:11]: can we have it as a "resolution" instead of a direct dependency?

Comment (TimoPeraza) [reply package.json:11]: Okey, great!

File: package.json [hunk 2]
```diff
@@ -8,10 +8,14 @@
...
     "turbolinks": "^5.2.0"
   },
   "version": "0.1.0",
   "devDependencies": {
     "webpack-dev-server": "^3.8.0"
+  },
+  "resolutions": {
+    "dot-prop": ">=5.2.0"
```
Comment (santib) [inline package.json:19]: I think this didn't took effect in the `yarn.lock` 🤔  maybe you missed to run `yarn install`?

**Comment by santib** [comment]
@TimoPeraza I tried downloading the branch locally and starting the server but couldn't. I'm getting:
```
➜  rails_api_base git:(fix/dot-prop-serialize-javascript) be rails s
warning: parser/current is loading parser/ruby26, which recognizes
warning: 2.6.5-compliant syntax, but you are running 2.6.6.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
=> Booting Puma
=> Rails 6.0.3.2 application starting in development
=> Run `rails server --help` for more startup options


========================================
  Your Yarn packages are out of date!
  Please run `yarn install --check-files` to update.
========================================


To disable this check, please change `check_yarn_integrity`
to `false` in your webpacker config file (config/webpacker.yml).


yarn check v1.17.0
success Folder in sync.
Done in 0.13s.
yarn check v1.17.0
error "@rails/webpacker#compression-webpack-plugin#serialize-javascript" is wrong version: expected "^2.1.2", got "4.0.0"
error "@rails/webpacker#optimize-css-assets-webpack-plugin#cssnano#cssnano-preset-default#postcss-merge-rules#postcss-selector-parser#dot-prop" is wrong version: expected "^4.1.1", got "5.2.0"
error Found 2 errors.
info Visit https://yarnpkg.com/en/docs/cli/check for documentation about this command.


Exiting
```

but if I run `yarn install --check-files` I get
```
➜  rails_api_base git:(fix/dot-prop-serialize-javascript) yarn install --check-files
yarn install v1.17.0
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning " > webpack-dev-server@3.8.0" has unmet peer dependency "webpack@^4.0.0".
warning "webpack-dev-server > webpack-dev-middleware@3.7.1" has unmet peer dependency "webpack@^4.0.0".
[4/4] 🔨  Building fresh packages...
✨  Done in 9.91s.
```

File: Gemfile [hunk 3]
```diff
@@ -25,7 +25,7 @@ gem 'pundit', '~> 2.1'
 gem 'rack-cors', '~> 1.0', '>= 1.0.6'
 gem 'sendgrid', '~> 1.2.4'
 gem 'sprockets', '~> 3.7.2'
-gem 'webpacker', '~> 4.2', '>= 4.2.2'
+gem 'webpacker', '~> 5.1', '>= 5.1.1'
```
Comment (santib) [inline Gemfile:28]: 👏

**Pull request merged**